xonsh/history/sqlite.py

```diff
@@ -220,6 +221,15 @@ def xh_sqlite_wipe_session(sessionid=None, filename=None):
         c.execute(sql, (str(sessionid),))
 
 
+def xh_sqlite_delete_input_matching(pattern, filename=None):
+    """Deletes entries from the database where the input matches a pattern."""
+    sql = f"DELETE FROM xonsh_history WHERE inp LIKE '{pattern}' ESCAPE '\\';"
```

Comment:
Here the like pattern is different from regex pattern right?

Reply:
Yes, I should probably rename it to `sql_pattern` or something similar to make it clear?

Reply:
what I meant was that the json implementation and this is not the same. `LIKE` operator doesn't implement the regex

Reply:
Ah, I think I understand :sweat_smile: 

When reading the sqlite documentation for [REGEXP](https://www.sqlite.org/lang_expr.html),
it says that there is no `regexp()`-function defined by default.

Is there anything in the `xonsh` code base that defines this, or is it fine to just assume that it is defined? :thinking:

Reply:
The base function can iterate over the entries and delete those matching the pattern. The sub-classes can implement an option to delete an item at the given index